src/probnum/diffeq/odefiltsmooth/ivpfiltsmooth.py

```diff
@@ -48,61 +49,49 @@ def solve(self, firststep, nsteps=1, **kwargs):
         ----------
         firststep : float
             First step for adaptive step size rule.
-        nsteps : int, optional
-            Number of inbetween steps for the filter. Default is 1.
         """
-
         ####### This function surely can use some code cleanup. #######
 
         current_rv = self.gfilt.initialrandomvariable
+        t = self.ivp.t0
+        times = [t]
+        rvs = [current_rv]
         step = firststep
         ssqest, num_steps = 0.0, 0
-        times, means, covars = [self.ivp.t0], [current_rv.mean()], [current_rv.cov()]
-        while times[-1] < self.ivp.tmax:
-            intermediate_step = float(step / nsteps)
-            t = times[-1]
-
-            pred_rv = current_rv
-            interms, intercs, interts = [], [], []
-            for idx in range(nsteps):
-                t_new = t + intermediate_step
-                pred_rv, __ = self.gfilt.predict(t, t_new, pred_rv, **kwargs)
-                interms.append(pred_rv.mean().copy())
-                intercs.append(pred_rv.cov().copy())
-                interts.append(t_new)
-                t = t_new
 
-            t_new = t
+        while t < self.ivp.tmax:
+
+            t_new = t + step
+            pred_rv, _ = self.gfilt.predict(t, t_new, current_rv, **kwargs)
+
             zero_data = 0.0
             filt_rv, meas_cov, crosscov, meas_mean = self.gfilt.update(
                 t_new, pred_rv, zero_data, **kwargs
             )
 
-            interms[-1] = filt_rv.mean().copy()
-            intercs[-1] = filt_rv.cov().copy()
-
             errorest, ssq = self._estimate_error(
                 filt_rv.mean(), crosscov, meas_cov, meas_mean
             )
 
             if self.steprule.is_accepted(step, errorest) is True:
-                times.extend(interts)
-                means.extend(interms)
-                covars.extend(intercs)
+                times.append(t_new)
+                rvs.append(filt_rv)
                 num_steps += 1
                 ssqest = ssqest + (ssq - ssqest) / num_steps
                 current_rv = filt_rv
+                t = t_new
 
             step = self._suggest_step(step, errorest)
```

Comment:
Make sure that we exactly hit tmax:
step = min(step, tmax-t)
if step < 1e-20:
    times[-1] = tmax
    break